master/internal/api_experiment.go

```diff
@@ -537,6 +538,11 @@ func (a *apiServer) CreateExperiment(
 
 var defaultMetricsStreamPeriod = 30 * time.Second
 
+func connectionIsClosed(resp grpc.ServerStream) bool {
```

Comment:
if you dont mind, change my stuff in `api_trials.go` to use this too?

Reply:
it's just a lot semantically clearer

Reply:
For sure

Reply:
So I added a similar implementation to logs.go - there ended up being a cyclic dependency if I reused code from (what is now called) grpcutil and this didn't seem worth starting a new generic "util" package. That code also just has the context, not the original GRPC stream, so in the original case we can make the code just a little cleaner too.

Sound good?